### PR TITLE
remove mb_scheduler, aka old kds, and related code

### PR DESCRIPTION
**Pull request by mamin506**
#### Problem solved by the commit
Code cleanup: Remove old KDS. 

#### What has been tested and how, request additional testing if necessary
xbutil validate/examine test on U50/U30 shells.

#### Documentation impact (if any)
No

This PR is to remove old kds code and some hacks from the xocl driver. But the utility tools will also needs to be changed, because all mb_scheduler sysfs nodes are deleted.

@rozumx @AShivangi , please take a look at the changes in query_requests.h, pcid/linux/device_linux.cpp and ReportDynamicRegion.cpp

@arpitxilinx @pranjalv-xilinx , Please take a look at pcie/windows/alveo/device_windows.cpp. Let me know if we need to talk about this offline.

@chvamshi-xilinx , the edge driver will be updated later. But there are some change in edge/user/shim.cpp and edge/user/device_linux.cpp. Please take a look.

### CODE REVIEW

**Review by maxzhen — APPROVED** [review]

**Review by chvamshi-xilinx — APPROVED** [review]

File: src/runtime_src/core/pcie/windows/alveo/device_windows.cpp [hunk 1]
```diff
@@ -1054,8 +1055,19 @@ struct kds_cu_info
 
     auto& stats = (*it).second;
     result_type cuStats;
-    for (unsigned int i = 0; i < stats->CuCount; i++)
-      cuStats.push_back(std::make_tuple(stats->CuInfo[i].BaseAddress, stats->CuInfo[i].Usage, 0));
+    for (unsigned int i = 0; i < stats->CuCount; i++) {
```
Comment (AShivangi) [inline src/runtime_src/core/pcie/windows/alveo/device_windows.cpp:1058]: Did you test this on Windows?

**Review by AShivangi — APPROVED** [review]
Looks good. Thanks for removing all the old code!

File: src/runtime_src/core/pcie/windows/alveo/device_windows.cpp (hunk 1 above)
Comment (mamin506) [reply src/runtime_src/core/pcie/windows/alveo/device_windows.cpp:1058]: I don't know how to test windows. :( 
I am working with @pranjalv-xilinx and @arpitxilinx to make sure this doesn't break windows build.

**Comment by mamin506** [comment]
@dkotipal-xilinx 
Hi Durga, this is a big change to remove old code that related to old KDS.
I would like you to take a look at the change in pcie/windows/alveo/device_windows.cpp.

Why:
In the old code, xbutil2 tool get CU name from ip_layout, but it get other cu information from query kds_cu_info. See the code I removed in ReportDynamicRegion.cpp.
In the new code, query kds_cu_info should contains cu index, name, address, usage, status. All of the info comes from driver.

What to do:
In windows build, stats->CuInfo[i] doesn't contain CU name. We need to add CU name into CuInfo[i]. @pranjalv-xilinx told me that you are the right person to work on this change.

The linux driver got CU address, name information in xclbin download. The windows driver can do the same.

File: src/runtime_src/core/pcie/windows/alveo/device_windows.cpp (hunk 1 above)
Comment (AShivangi) [reply src/runtime_src/core/pcie/windows/alveo/device_windows.cpp:1058]: Yeah. Let's test it on windows before merging the code. Also, compare the old to new outputs to make sure all the information is correctly displayed

**Review by stsoe — CHANGES_REQUESTED** [review]
This looks like a significant change.  I am not sure how much was dead code and how much is actually change.  But testing only xbutil validate does not sound sufficient.  Can you run some DSV board test suites before publishing?

File: src/runtime_src/core/pcie/windows/alveo/device_windows.cpp [hunk 2]
```diff
@@ -1054,8 +1055,19 @@ struct kds_cu_info
...
+      data_type data;
```
Comment (rozumx) [inline src/runtime_src/core/pcie/windows/alveo/device_windows.cpp:1059]: Initialize the structure to a zeros.
```suggestion
      data_type data = {};
```

**Review by rozumx — CHANGES_REQUESTED** [review]
Overall, I like this removal of code for it helps simplify the code base and makes it easier to support.  

Please do some testing on both Linux and Windows to validate correct functionality and that the code doesn't regress.  Once completed, I will approve this PR.

**Comment by stsoe** [comment]
> This looks like a significant change. I am not sure how much was dead code and how much is actually change. But testing only xbutil validate does not sound sufficient. Can you run some DSV board test suites before publishing?

I validated this PR with old U200 HW board tests, which all completed as expected.

**Review by stsoe — APPROVED** [review]
Validated PR with U200 board tests.

File: src/runtime_src/core/pcie/windows/alveo/device_windows.cpp (hunk 2 above)
Comment (mamin506) [reply src/runtime_src/core/pcie/windows/alveo/device_windows.cpp:1059]: Thanks Stephen for the comment. I will leave this change to Windows guys. They will need to update the CU name line.

**Review by arpitxilinx — APPROVED** [review]

File: src/runtime_src/core/pcie/windows/alveo/device_windows.cpp (hunk 2 above)
Comment (mamin506) [reply src/runtime_src/core/pcie/windows/alveo/device_windows.cpp:1059]: @rozumx Soren also help test some cases on his side. It looks good.
I have talk to the Windows guys. They said we can go ahead and merge this change.

I will let Max know and merge the change.

**Review by rozumx — APPROVED** [review]

**Pull request merged**